#!/bin/sh /usr/share/dpatch/dpatch-run
## 04-vboxdrv-references.dpatch by Daniel Hahler <ubuntu@thequod.de>
##
## DP: Change error messages for vboxdrv failure, according to
## DP: virtualbox-ose-modules.

@DPATCH@

diff -Naurp virtualbox-ose-1.6.0-dfsg.orig/src/VBox/VMM/VM.cpp virtualbox-ose-1.6.0-dfsg/src/VBox/VMM/VM.cpp
--- virtualbox-ose-1.6.0-dfsg.orig/src/VBox/VMM/VM.cpp	2008-04-30 14:05:26.000000000 +0000
+++ virtualbox-ose-1.6.0-dfsg/src/VBox/VMM/VM.cpp	2008-05-09 09:25:54.000000000 +0000
@@ -257,10 +257,8 @@ VMR3DECL(int)   VMR3Create(PFNVMATERROR 
             {
                 case VERR_VM_DRIVER_LOAD_ERROR:
 #ifdef RT_OS_LINUX
-                    pszError = N_("VirtualBox kernel driver not loaded. The vboxdrv kernel module "
-                                  "was either not loaded or /dev/vboxdrv is not set up properly. "
-                                  "Re-setup the kernel module by executing "
-                                  "'/etc/init.d/vboxdrv setup' as root");
+                    pszError = N_("Please install the virtualbox-ose-modules package for your "
+		    		  "kernel and execute 'modprobe vboxdrv' as root");
 #else
                     pszError = N_("VirtualBox kernel driver not loaded");
 #endif
@@ -280,10 +278,8 @@ VMR3DECL(int)   VMR3Create(PFNVMATERROR 
                     break;
                 case VERR_VM_DRIVER_NOT_INSTALLED:
 #ifdef RT_OS_LINUX
-                    pszError = N_("VirtualBox kernel driver not installed. The vboxdrv kernel module "
-                                  "was either not loaded or /dev/vboxdrv was not created for some "
-                                  "reason. Re-setup the kernel module by executing "
-                                  "'/etc/init.d/vboxdrv setup' as root");
+                    pszError = N_("Please install the virtualbox-ose-modules package for your "
+		    		  "kernel and execute 'modprobe vboxdrv' as root");
 #else
                     pszError = N_("VirtualBox kernel driver not installed");
 #endif
